### PR TITLE
Simon bipi 2125

### PR DESCRIPTION
**Pull request by simongrac**
Adding an IP address to the service /phoxi_camera/get_device_list output, based on avahi-browse call. After this terminal call, output is parsed and IP addresses of available scanners are obtained. IP addresses are added to the service output then. 

This PR is connected with https://github.com/photoneo/binpicking/pull/267 and https://github.com/photoneo/phoros_binpicking/pull/80 - have to be merged SIMULTANEOUSLY

### CODE REVIEW

File: src/PhoXiInterface.cpp [hunk 1]
```diff
@@ -11,6 +11,14 @@ namespace phoxi_camera {
 
     }
 
+    // scanners IP obtaining methods
+    int countRowsWithStartingSign(char, std::vector<std::string>);
```
Comment (JakubHazik) [inline src/PhoXiInterface.cpp]: Add some function description. Check descriptions in PhoXIinterface.h

File: src/PhoXiInterface.cpp [hunk 2]
```diff
@@ -317,4 +336,120 @@ namespace phoxi_camera {
...
+    std::map<std::string, std::string> getScannersIPs() {
+        std::map<std::string, std::string> scannersIPs;
+        const char *command = "avahi-browse -r -t _3d-camera._tcp";
+        char buffer[256];
+        std::string result;
+        FILE *pipe = popen(command, "r");
+
+        std::vector<std::string> resultByRows;
+        if (!pipe) {
+            throw std::runtime_error("avahi-browse operation failed!");
```
[the remark this answers is not in the document]
Comment (simongrac) [reply src/PhoXiInterface.cpp]: has sense, done

File: src/PhoXiInterface.cpp [hunk 3]
```diff
@@ -317,4 +336,120 @@ namespace phoxi_camera {
...
+        }
+        try {
+            while (fgets(buffer, sizeof buffer, pipe) != nullptr) {
+                resultByRows.emplace_back(buffer);
+            }
+        } catch (...) {
```
[the remark this answers is not in the document]
Comment (simongrac) [reply src/PhoXiInterface.cpp]: changed for ferror checking

File: src/PhoXiInterface.cpp (hunk 1 above)
Comment (simongrac) [reply src/PhoXiInterface.cpp]: moved to PhoXIinterface.h, descriptions added

File: src/PhoXiInterface.cpp [hunk 4]
```diff
@@ -19,6 +20,17 @@ namespace phoxi_camera {
         std::vector<PhoXiDeviceInformation> deviceInfo;
         auto dl = phoXiFactory.GetDeviceList();
         toPhoXiCameraDeviceInforamtion(dl, deviceInfo);
+
+        std::map<std::string, std::string> scannersIPs = PhoXiInterface::getScannersIPs();
+
+        for (auto &device : deviceInfo) {
+            if (scannersIPs.empty() || (scannersIPs.find(device.hwIdentification) == scannersIPs.end())) {
+                device.IPaddress = "unknown";
```
Comment (martinstofanak) [inline src/PhoXiInterface.cpp]: This string ("unknown") is in "binpicking" repo forwarded to the dropdown list UI, so in current implementation its not translated. Is it ok, in this case? @simongrac @najlepsiwebdesigner

**Pull request merged**